### ISSUE TITLE
Roadmap to v2.0.0

### ISSUE THREAD

**Issue by clue** [clue/phar-composer#88]
It's my understanding that this project seems to work reasonably well for its intended use case - and at the same time fails to implement many of the nice-to-have features that were added over time.

As such, I'm planning to work towards a breaking v2.0.0 release to bring this project back to its basics: Building phar files for any project with as little configuration as possible.

In particular, this means that it will focus on building only and will not have any of the additional search and install commands anymore. This way, we can remove a significant piece of the current code base, no longer require any command line invocations (no composer, no git, no nothing) and focus on building the best tool to build a phar from a project directory.

On top of this, we may provide some documentation on how to build arbitrary projects by Composer package name or git repository, most of which can be achieved with a single line bash script.

I'm opening this ticket for the reference and to hopefully spark some discussion. I'm still working towards creating a feature/maintenance release for the v1 series, but future development will focus just on building phars.

**Comment by ppetermann** [comment]
Mh, that's a bit of a bummer, being able to install tools easily was my main use case for phar-composer and what I've been recommending it for over the years. 

However that opens to me the question: if basically you are reducing the features to the same things Box wants to do, wouldn't it make sense to bundle efforts with them?

**Comment by ppetermann** [comment]
Nevermind, box seems to be stale

**Comment by clue** [comment]
@ppetermann Good point, this is definitely something to consider for the upcoming version! It looks like @theofidry has done an outstanding job of updating Box and improving it significantly since the last time I checked: https://github.com/humbug/box

Box used to rely heavily on explicit configuration, which was part of the motivation for creating phar-composer. Nowadays, they seem to be much more similar.

Perhaps it will make more sense to join efforts and deprecate one of the projects in the future, I'm open for any input and don't feel maintaining two very similar projects is a good time investment in the long run :+1: Ideally, another project may focus on the "higher level" decisions like bundling arbitrary projects by name instead of just project directories, installing and managing them system-wide etc. Let me know what you think :+1:

**Comment by theofidry** [comment]
Thanks for the mention :)

I would be more than happy to receive help for Box. I did some efforts to allow to make it work as easily as possible with most projects, although there might be some points to improve still.

That said I admit I think the next big points of improvements are rather around the PHAR tooling than the PHAR bundling itself:

- [PHP-Scoper](https://github.com/humbug/php-scoper) cf. [PHAR code isolation](https://github.com/humbug/box/blob/master/doc/code-isolation.md#phar-code-isolation)
- Better usability with the `box diff` command (which heavily relies on [pharaoh](https://github.com/paragonie/pharaoh)
- Possibility to go a step further and try to ship a true binary instead of a PHP app, see https://github.com/humbug/box/issues/331
- Provide an easier way to distribute PHARs compatible with an old codebase without forcing the codebase to stay on old PHP versions: https://github.com/humbug/box/issues/330
- Make it easier to sign a PHAR: https://github.com/humbug/box/issues/329 https://github.com/humbug/box/issues/15
- Make it easier to install a PHAR: https://github.com/humbug/box/issues/8
- Speed up the PHAR bundling when using PHP-Scoper: https://github.com/humbug/box/pull/429
- Consolidating the codebase like helping out with the new version of https://github.com/doctrine/annotations

So although the idea of bundling a PHAR might seem to be a fun thing to do, there is still a lot to do afterwards! 😄
